src/main/html/index.html

```diff
@@ -74,6 +74,15 @@
         obj.method + '&' +
         encodeURIComponent(url) + '&' +
         encodeURIComponent(params.join('&'));
+
+      // handle URL encoding of parameters that are not encoded by the encodeURIComponent
+      // method to ensure compliance with RFC 3986
+      to_sign = to_sign.replace(/\*/g, "%2A")
+                       .replace(/\!/g, "%21")
+                       .replace(/\'/g, "%27")
+                       .replace(/\(/g, "%28")
+                       .replace(/\)/g, "%29");
```

Comment:
Looks good to me! This is almost identical to oauth_reverse_proxy's encoder: https://github.com/Cimpress-MCP/oauth_reverse_proxy/blob/master/lib/encoding.js#L7

@ryanbreen can you merge?

At the risk of playing code golf, I will point out this RFC 3986 / 5987 snippet from MDN docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#Examples. In this PR there are 5 RegExes in a row; if there were any more then I'd strongly advise combining them and using a var so that really long strings don't make the browser work too hard.